command_line/generate_tutorial_text.py

```diff
@@ -21,33 +21,34 @@ def _command_runner(
     """Run a command and write its output to a defined location"""
     print(" ".join(str(e) for e in command))
     if output_directory and store_command is None:
-        store_command = output_directory.join(command[0] + ".cmd")
+        store_command = output_directory / (command[0] + ".cmd")
     if output_directory and store_output is None:
-        store_output = output_directory.join(command[0] + ".log")
+        store_output = output_directory / (command[0] + ".log")
     if store_command:
-        store_command.write(" ".join(str(e) for e in command), ensure=True)
+        store_command.write_text(" ".join(str(e) for e in command))
```

Comment:
```suggestion
        store_command.parent.mkdir(parents=True, exist_ok=True)
        store_command.write_text(" ".join(str(e) for e in command))
```

Technically this is required to match the previous behaviour (although it _probably_ doesn't matter?)